Sources/StreamChatUI/Composer/ComposerVC.swift

```diff
@@ -794,3 +819,77 @@ open class ComposerVC: _ViewController,
         }
     }
 }
+
+/// searchUsers does an autocomplete search on a list of ChatUser and returns users with `id` or `name` containing the search string
+/// results are returned sorted by their edit distance from the searched string
+/// distance is calculated using the levenshtein algorithm
+/// both search and name strings are normalized (lowercased and by replacing diacritics)
+func searchUsers(_ users: [ChatUser], by searchInput: String, excludingId: String? = nil) -> [ChatUser] {
+    let normalize: (String) -> String = {
+        $0.lowercased().folding(options: .diacriticInsensitive, locale: .current)
+    }
+
+    let searchInput = normalize(searchInput)
+
+    let matchingUsers = users.filter { $0.id != excludingId }
+        .filter { $0.id.contains(searchInput) || (normalize($0.name ?? "").contains(searchInput)) }
+
+    let uniqueUsers = matchingUsers.reduce(into: [String: ChatUser]()) {
+        $0[$1.id] = $1
+    }
+
+    let distance: (ChatUser) -> Int = {
+        min($0.id.levenshtein(searchInput), $0.name?.levenshtein(searchInput) ?? 1000)
+    }
+
+    return uniqueUsers.values.map { $0 }.sorted {
+        distance($0) < distance($1)
+    }
+}
+
+extension String {
+    subscript(index: Int) -> Character {
+        self[self.index(startIndex, offsetBy: index)]
+    }
+}
+
+extension String {
+    public func levenshtein(_ other: String) -> Int {
+        let sCount = count
+        let oCount = other.count
+
+        guard sCount != 0 else {
+            return oCount
+        }
+
+        guard oCount != 0 else {
+            return sCount
+        }
+
+        let line: [Int] = Array(repeating: 0, count: oCount + 1)
+        var mat: [[Int]] = Array(repeating: line, count: sCount + 1)
+
+        for i in 0...sCount {
+            mat[i][0] = i
+        }
+
+        for j in 0...oCount {
+            mat[0][j] = j
+        }
+
+        for j in 1...oCount {
+            for i in 1...sCount {
+                if self[i - 1] == other[j - 1] {
+                    mat[i][j] = mat[i - 1][j - 1] // no operation
+                } else {
+                    let del = mat[i - 1][j] + 1 // deletion
+                    let ins = mat[i][j - 1] + 1 // insertion
+                    let sub = mat[i - 1][j - 1] + 1 // substitution
+                    mat[i][j] = min(min(del, ins), sub)
+                }
+            }
+        }
+
+        return mat[sCount][oCount]
+    }
+}
```

Comment:
I think all these should live in the `UserSearchController`. And since the levenshtein algorithm can be used in other types of searches, it is worth creating the extension in a separate file, currently all our string extensions are in "Utils/String+Extensions.swift", and the test of the algorithm should also live in "Utils/String+Extensions_Tests.swift"